offchain/rollups-events/src/broker/mod.rs

```diff
@@ -223,9 +221,34 @@ impl Broker {
         event.try_into()
     }
 
+    /// Consume the next event in stream
+    ///
+    /// This function blocks until a new event is available,
+    /// and it retries timeouts instead of returning an error.
+    ///
+    /// To consume the first event in the stream, `last_consumed_id` should be `INITIAL_ID`.
+    #[tracing::instrument(level = "trace", skip_all)]
+    pub async fn consume_blocking<S: BrokerStream>(
+        &mut self,
+        stream: &S,
+        last_consumed_id: &str,
+    ) -> Result<Event<S::Payload>, BrokerError> {
+        loop {
+            let result = self
+                .consume_blocking_deprecated(stream, last_consumed_id)
+                .await;
+
+            if let Err(BrokerError::ConsumeTimeout) = result {
+                tracing::trace!("consume timed out, retrying");
+            } else {
+                return result;
+            }
+        }
+    }
```

Comment:
> I'm quite confused about this code. Isn't the loop basically ignoring the timeout configuration by trying all over again? If so, why the change in behavior? I feel I might be missing something here.
>
> I also didn't understand why you "deprecated" the original function but used it internally nevertheless. You did this to avoid dealing with the possible side-effects of changing the behavior in its callers in this PR?

Yes, I guess so but @gligneul can answer it better, I believe.

Reply:
The idea of the timeout is to avoid hanging the Redis connection. So, instead of telling Redis to wait forever, the code sets a timeout and tries again immediately.

This behavior was being implemented by the broker clients manually. If you check the advance-runner code, for instance, you will see that it also immediately retries after the timeout (https://github.com/cartesi/rollups-node/blob/main/offchain/advance-runner/src/broker.rs#L107).

Talking with Renan, we decided that since it is the default intended behavior, we should move it to inside the library instead of re-implementing the same thing in every client. So, the new function implements the desired behavior and hides this timeout/retry logic.

The deprecated function should be made internal, and we need to remove the timeout handling from the other clients and use the new function to do that. @marcelstanley, can you evaluate if it is doable in this PR?

Reply:
> The deprecated function should be made internal, and we need to remove the timeout handling from the other clients and use the new function to do that. @marcelstanley, can you evaluate if it is doable in this PR?

Sure thing, @gligneul.

Reply:
Done at b60f6d6b